cabby/geo/walk.py

```diff
@@ -219,24 +286,22 @@ def get_start_poi(self,
 
     poi_in_ring = self.map.poi[self.map.poi['osmid'].isin(osmid_in_range)]
 
-    # Filter with name.
-    named_poi = poi_in_ring[poi_in_ring['name'].notnull()]
-
     # Filter large POI.
-    small_poi = named_poi[named_poi['s2cellids'].str.len() <= SMALL_POI]
+    small_poi = poi_in_ring[poi_in_ring['s2cellids'].str.len() <= SMALL_POI]
 
-    if small_poi.shape[0] == 0:
+    # Filter non-specific tags.
+    start_point = self.select_non_specific_poi(small_poi)
+    
+    if start_point is None:
       return None
 
-    # Pick random POI.
-    start_point = self.sample_point(small_poi)
     start_point['geometry'] = start_point.centroid
```

Comment:
This duplicates logic from get_end_poi(). Can you reduce the redundancy?

Reply:
reduced duplication